src/lib/extend.ts

```diff
@@ -146,5 +149,75 @@ const extend = {
     },
 };
 
+const modernExtend = {
+    enumLookup: (args: {
+        name: string, lookup: KeyValue, cluster: string | number, attribute: string | {id: number, type: number}, description: string,
+        zigbeeCommandOptions?: {manufacturerCode: number}, readOnly?: boolean,
+    }): ModernExtend => {
+        const {name, lookup, cluster, attribute, description, zigbeeCommandOptions} = args;
+        const attributeKey = isString(attribute) ? attribute : attribute.id;
+        const expose = new Enum(name, args.readOnly ? access.STATE_GET : access.ALL, Object.keys(lookup)).withDescription(description);
+        const fromZigbee: Fz.Converter[] = [{
+            cluster,
+            type: ['attributeReport', 'readResponse'],
+            convert: (model, msg, publish, options, meta) => {
+                if (attributeKey in msg.data) {
+                    return {[expose.property]: getFromLookupByValue(lookup, msg.data[attributeKey])};
+                }
+            },
+        }];
+        const toZigbee: Tz.Converter[] = [{
+            key: [name],
+            convertSet: args.readOnly ? undefined : async (entity, key, value, meta) => {
+                const payloadValue = getFromLookup(value, lookup);
+                const payload = isString(attribute) ? {[attribute]: payloadValue} : {[attribute.id]: {value: payloadValue, type: attribute.type}};
+                await entity.write(cluster, payload, zigbeeCommandOptions);
+                return {state: {[key]: value}};
+            },
+            convertGet: async (entity, key, meta) => {
+                // @ts-expect-error TODO fix zh type
+                await entity.read(cluster, [attributeKey], zigbeeCommandOptions);
+            },
+        }];
+        return {exposes: [expose], fromZigbee, toZigbee, isModernExtend: true};
+    },
+    numeric: (args: {
+        name: string, cluster: string | number, attribute: string | {id: number, type: number}, description: string,
+        zigbeeCommandOptions?: {manufacturerCode: number}, readOnly?: boolean, unit?: string,
+    }): ModernExtend => {
+        const {name, cluster, attribute, description, zigbeeCommandOptions} = args;
+        const attributeKey = isString(attribute) ? attribute : attribute.id;
+
+        let expose = new Numeric(name, args.readOnly ? access.STATE_GET : access.ALL).withDescription(description);
+        if (args.unit) expose = expose.withUnit(args.unit);
+
+        const fromZigbee: Fz.Converter[] = [{
+            cluster,
```

Comment:
For some reason (not sure why), if `cluster` is a number here, the inbound messages from the Sonoff TRV do not get processed.  If it is a string representation of the decimal cluster ID, however, then it works as expected. Any ideas why this might be the case?

For example, in `fzLocal`:

...the following works:

```
    child_lock: {
        cluster: '64529',
```
whereas the following does not:

```
    child_lock: {
        cluster: 0xFC11,
```

Reply:
If I change the `extend.numeric` to the following, then all the attributes in #6469 are populated correctly:

```        
const fromZigbee: Fz.Converter[] = [{
      cluster: cluster.toString(),
```

Reply:
If I use the string representation in the extend definition, as follows:

```
extend.numeric({
                name: 'idle_steps',
                cluster: '64529',
                attribute: {id: 0x6003, type: 0x21},
                description: 'Number of steps used for calibration (no-load steps)',
                readOnly: true,
            })
``` 
then I get the following error:

<img width="343" alt="image" src="https://github.com/Koenkk/zigbee-herdsman-converters/assets/1983638/e013ee51-2268-48ee-9e8f-d12093fc6bbd">

So I need to use a numeric cluster ID when sending messages to the device, but a decimal string when handling the incoming messages. (See my previous `'tzLocal` and `fzLocal` examples in `sonoff.ts`).

Reply:
Fixed!

Reply:
I can confirm this now works. Raised #6559 which would allow me to move the frost protection into the modern numeric extend.

Reply:
@photomoose for the value scaling, you can use the `scale` parameter which was added by @kirovilya  in https://github.com/Koenkk/zigbee-herdsman-converters/pull/6554

Reply:
Nice, `scale` will work in this case, I'll use that once the PR is merged.

Reply:
Raised #6566 for some of the Sonoff attributes.